sklearn/preprocessing/data.py

```diff
@@ -2440,3 +2446,189 @@ def quantile_transform(X, axis=0, n_quantiles=1000,
     else:
         raise ValueError("axis should be either equal to 0 or 1. Got"
                          " axis={}".format(axis))
+
+
+class CategoricalEncoder(BaseEstimator, TransformerMixin):
+    """Encode categorical features using a one-hot aka one-of-K scheme.
+
+    The input to this transformer should be a matrix of integers or strings,
+    denoting the values taken on by categorical (discrete) features. The
+    output will be a sparse matrix where each column corresponds to one
+    possible value of one feature.
+
+    This encoding is needed for feeding categorical data to many scikit-learn
+    estimators, notably linear models and SVMs with the standard kernels.
+    Read more in the :ref:`User Guide <preprocessing_categorical_features>`.
+
+    Parameters
+    ----------
+    classes : 'auto', 2D array of ints or strings or both.
+        Values per feature.
+
+        - 'auto' : Determine classes automatically from the training data.
+        - array: ``classes[i]`` holds the classes expected in the ith column.
```

Comment:
Does this specify the *order* of these values in the output, or just the set? Must be clear.